tests/Unit/Helpers/ControlSystem/Examples.hpp

```diff
@@ -0,0 +1,137 @@
+  static void apply(const gsl::not_null<db::DataBox<DbTags>*> box,
+                    Parallel::GlobalCache<Metavariables>& /*cache*/,
+                    const ArrayIndex& /*array_index*/, const double time,
+                    tuples::TaggedTuple<SubmeasurementTag> data) noexcept {
+    if constexpr (db::tag_is_retrievable_v<::Tags::Time, db::DataBox<DbTags>>) {
+      db::mutate<::Tags::Time, MeasurementResultTag>(
+          box,
+          [&time, &data](const gsl::not_null<double*> stored_time,
+                         const gsl::not_null<double*> stored_data) noexcept {
+            *stored_time = time;
+            *stored_data = get<SubmeasurementTag>(data);
+          });
+    } else {
+      ERROR("Called with the wrong DataBox");
+    }
+  }
+};
+
+/// [Submeasurement]
+struct ExampleSubmeasurement
+    : tt::ConformsTo<control_system::protocols::Submeasurement> {
+  // This submeasurement does not use an interpolation component.
+  template <typename ControlSystems>
+  using interpolation_target_tag = void;
+
+  using argument_tags = tmpl::list<SomeTagOnElement>;
+
+  template <typename Metavariables, typename ParallelComponent,
+            typename ControlSystems>
+  static void apply(
+      const double data_from_element,
+      const LinkedMessageId<double>& measurement_id,
+      Parallel::GlobalCache<Metavariables>& cache, const int& /*array_index*/,
+      const ParallelComponent* /*meta*/, ControlSystems /*meta*/) noexcept {
+    // In real cases, we would generally do a reduction to a single
+    // chare here and have the below code in the reduction action, but
+    // the action testing framework doesn't support reductions, so
+    // this example is for a control system running on a singleton and
+    // we just pass the data through.
+    const auto box =
+        db::create<db::AddSimpleTags<MeasurementResultTag>>(data_from_element);
+    control_system::RunCallbacks<ExampleSubmeasurement, ControlSystems>::apply(
+        box, cache, measurement_id);
+  }
+};
+/// [Submeasurement]
+
+/// [Measurement]
+struct ExampleMeasurement
+    : tt::ConformsTo<control_system::protocols::Measurement> {
+  using submeasurements = tmpl::list<ExampleSubmeasurement>;
+};
+/// [Measurement]
+
+/// [ControlSystem]
+struct ExampleControlSystem
+    : tt::ConformsTo<control_system::protocols::ControlSystem> {
+  static std::string name() noexcept { return "ExampleControlSystem"; }
+  using measurement = ExampleMeasurement;
+
+  // This is not part of the required interface, but is used by this
+  // control system to store the measurement data.  Most control
+  // systems will do something like this.
+  struct ExampleSubmeasurementQueueTag {
+    using type = double;
+  };
+
+  // As with the previous struct, this is not part of the required
+  // interface.
+  struct MeasurementQueue : db::SimpleTag {
+    using type =
+        LinkedMessageQueue<double, tmpl::list<ExampleSubmeasurementQueueTag>>;
+  };
```

Comment:
One thing I found useful was adding a type alias `simple_tags` to the control system so that the control component could easily access this queue to put in its databox. Also not required for the interface but may be useful for the example.

Reply:
If it's used by the component it should be part of the specified interface.  We can decide on that when the protocol is expanded for the component stuff.